tests/generic/core/test_SerialConsole.py

```diff
@@ -314,3 +317,23 @@ def test_SerialConsole_login_except_on_wrong_username_match(serial_console_proxy
 
 def test_SerialConsole_does_not_require_login(serial_console_proxy):
     assert serial_console_proxy.console.requires_login is True
+
+
+def test_SerialConsole_raw_log_file(pty_pair):
+    received = 'abcd\nefgh'
+
+    with tempfile.NamedTemporaryFile(delete=False) as tmpfile:
```

Comment:
I don't think you need to keep the file around outside the test, feel free to correct me if I'm wrong.
```suggestion
    with tempfile.NamedTemporaryFile() as tmpfile:
```